gravitee-am-service/src/main/java/io/gravitee/am/service/impl/PasswordPolicyServiceImpl.java

```diff
@@ -83,20 +81,26 @@ public Flowable<PasswordPolicy> findByDomain(String domain) {
                 });
     }
 
+
     @Override
     public Single<PasswordPolicy> create(PasswordPolicy policy, User principal) {
         log.debug("Create a new password policy named '{}' for {} {}", policy.getName(), policy.getReferenceType(), policy.getReferenceId());
-        // TODO during AM-2893, check if there is existing policies, if not set this one as default
 
         if (!hasLength(policy.getReferenceId()) && !ReferenceType.DOMAIN.equals(policy.getReferenceType())) {
-            return Single.error(new InvalidParameterException("Password policy requires a reference type and a reference ID"));
+            return Single.error(new InvalidParameterException("Password policy requires a reference type DOMAIN and a reference ID"));
         }
 
         final var now = new Date();
         policy.setCreatedAt(now);
         policy.setUpdatedAt(now);
 
-        return passwordPolicyRepository.create(policy)
+        return passwordPolicyRepository.findByDefaultPolicy(policy.getReferenceType(), policy.getReferenceId())
+                .flatMap(__ -> Maybe.just(policy))
```

Comment:
```suggestion
                .toMaybe()
```
?

Reply:
Here we dont want to operate on Maybe returned by findByDefaultPolicy. We want to operate on policy from method argument. Thats why we create maybe and then we proceed this in stream. `toMaybe()` will create stream with object from DB.

Reply:
oh, I can see that Maybe is already return so this line can be remove at all.